### PR TITLE
Install `ktlint` linter and fix lint issues

### PR DESCRIPTION
**Pull request by FloEdelmann**
Follow-up to #3649.

This PR installs the [ktlint](https://github.com/pinterest/ktlint) linter using the [ktlint-check](https://github.com/musichin/ktlint-check) GitHub Action. It runs everytime commits are pushed to the master branch, and indicates its result with a :heavy_check_mark: or :x: next to the commit.

Only the rules accepted in https://github.com/streetcomplete/StreetComplete/pull/3649#issuecomment-1015692558 are enabled. I fixed the lint issues reported by all these rules (in individual commits).

Additionally, I also fixed some lint issues reported by the rejected rules (in individual commits), but each change was reviewed if it would actually improve readability.

On top of that, I did some more manual lint changes (in individual commits) to improve consistency and readability.

### CODE REVIEW

**Comment by westnordost** [comment]
This is a hell of a PR to go through, I think I'll rather review+merge it after releasing v40.0-beta1 which I wanted to do today

**Comment by smichel17** [comment]
> This is a hell of a PR to go through

Does it make any sense to break it up into chunks? First merge the linter, then trickle in PRs enabling the rules at a pace that's reasonable to review.

File: app/src/main/java/de/westnordost/streetcomplete/util/CrashReportExceptionHandler.kt [hunk 1]
```diff
@@ -56,12 +56,12 @@ import javax.inject.Singleton
...
         val report = """
-Describe how to reproduce it here:
+            Describe how to reproduce it here:
 
 
 
-$error
-"""
+            $error
+        """.trimIndent()
```
[the remark this answers is not in the document]
Comment (FloEdelmann) [reply app/src/main/java/de/westnordost/streetcomplete/util/CrashReportExceptionHandler.kt]: Oh, you're right! I misunderstood how `trimIndent` works (I thought that it removes all leading whitespace in every line).

**Review by westnordost — APPROVED** [review]

**Comment by FloEdelmann** [comment]
Yay, thanks for merging!

I now found a ktlint plugin for IntelliJ IDEA which shows lint issues inline in the code: https://plugins.jetbrains.com/plugin/15057-ktlint-unofficial-

Is there some place/documentation where we can recommend installing it?

**Comment by matkoniecz** [comment]
https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING.md#development ?

**Comment by FloEdelmann** [comment]
Done in https://github.com/streetcomplete/StreetComplete/commit/88d71c4a792376d82d5b1863553cf96f551088f7.

**Pull request merged**